src/Microsoft.TemplateEngine.TemplateLocalizer/TemplateJsonProvider.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.IO;
+
+namespace Microsoft.TemplateEngine.TemplateLocalizer
+{
+    internal sealed class TemplateJsonProvider
+    {
+        /// <summary>
+        /// Creates an instance of <see cref="TemplateJsonProvider"/>.
+        /// </summary>
+        /// <param name="path">A path to a template.jsonfile or a path to a directory that contains a template.json file.</param>
+        public TemplateJsonProvider(string path)
+        {
+            Path = path;
```

Comment:
what if path is null or inexistent?

Reply:
That is OK. Validating and making sense of the user provided value is done later in the instance methods of this type.